src/hal/src/device.rs

```diff
@@ -758,4 +758,42 @@ pub trait Device<B: Backend>: fmt::Debug + Any + Send + Sync {
     ///
     /// Host access to all queues needs to be **externally** sycnhronized!
     fn wait_idle(&self) -> Result<(), OutOfMemory>;
+
+    /// Associate a name with an image, for easier debugging in external tools or with validation
+    /// layers that can print a friendly name when referring to objects in error messages
+    unsafe fn set_image_name(&self, image: &B::Image, name: &str) -> Result<(), OutOfMemory>;
```

Comment:
I think we should just ignore the errors here. They aren't useful, they just make a lot of noise.

Reply:
I'd be inclined to follow the Vulkan specification and keep the errors here, just to be consistent with the rest of the API, even if most of the time the user would not care.

Reply:
We follow Vulkan for the lack of a better idea most of the time. Sometimes though, we have a better idea, and we disagree. See https://github.com/gfx-rs/gfx/wiki/Deviations-from-Vulkan as relevant.

In this particular case, labels are most likely in the host memory. Running out of host memory is something that can easily happen in 99% of the gfx-rs code, given our use of the standard library and containers, and we aren't handling that.

So for now, let's not worry about these error codes. We'll cross that bridge when it will need to be crossed.

Reply:
I've removed the error codes.